destreamer.ts

```diff
@@ -1,226 +1,223 @@
+    username: { type: 'string', demandOption: false },
+    outputDirectory: { type: 'string', alias: 'outputdirectory', default: 'videos' },
+    format: {
+        alias: 'f',
+        describe: `Expose youtube-dl --format option, for details see\n
+        https://github.com/ytdl-org/youtube-dl/blob/master/README.md#format-selection`,
+        type: 'string',
+        demandOption: false
+    },
+    simulate: {
+        alias: 's',
+        describe: `If this is set to true no video will be downloaded and the script
+        will log the video info (default: false)`,
+        type: 'boolean',
+        default: false,
+        demandOption: false
+    },
+}).argv;
+
+if (argv.simulate){
+    console.info('Video URLs: %s', argv.videoUrls);
+    console.info('Username: %s', argv.username);
+    console.info(colors.green('There will be no video downloaded, it\'s only a simulation\n'));
+} else {
+    console.info('Video URLs: %s', argv.videoUrls);
+    console.info('Username: %s', argv.username);
+    console.info('Output Directory: %s', argv.outputDirectory);
+    console.info('Video/Audio Quality: %s', argv.format);
+}
+
+
+function sanityChecks() {
+    try {
+        const ytdlVer = execSync('youtube-dl --version');
+        console.info(colors.green(`Using youtube-dl version ${ytdlVer}`));
+    }
+    catch (e) {
+        console.error('You need youtube-dl in $PATH for this to work. Make sure it is a relatively recent one, baked after 2019.');
+        process.exit(22);
+    }
+
+    try {
+        const ffmpegVer = execSync('ffmpeg -version')
+            .toString().split('\n')[0];
+        console.info(colors.green(`Using ${ffmpegVer}\n`));
+    }
+    catch (e) {
+        console.error('FFmpeg is missing. You need a fairly recent release of FFmpeg in $PATH.');
+    }
+
+    if (!fs.existsSync(argv.outputDirectory)){
+        console.log('Creating output directory: ' +
+            process.cwd() + path.sep + argv.outputDirectory);
+        fs.mkdirSync(argv.outputDirectory);
+    }
+}
+
+
+async function DoInteractiveLogin(username?: string): Promise<Session> {
+    console.log('Launching headless Chrome to perform the OpenID Connect dance...');
+    const browser = await puppeteer.launch({
+        headless: false,
+        args: ['--disable-dev-shm-usage']
+    });
+    const page = (await browser.pages())[0];
+    console.log('Navigating to microsoftonline.com for login procedure...');
+
+    await page.goto(loginUrl, { waitUntil: "load" });
+    await page.waitForSelector('input[type="email"]');
+    if (username){
+        await page.keyboard.type(username);
+        await page.click('input[type="submit"]');
+    }
+
+    // Im leaving this so taht we can find out when the user finishes the login procedure
+    // so that we don't try and evaluate the session for 2 and a half minutes
+    await browser.waitForTarget(target => (
+        target.url().includes("microsoftstream.com") &&
+        target.url().endsWith("/")
```

Comment:
Should we test for noSignUpCheck instead? What if later on a different default query parameter is introduced (like feature flags), i think this will fail in that scenario.

Reply:
I might just had the right idea: since the first response is just js, and the DOM loads only after the redirect we can wait for something like the logo element to load

Reply:
But why not just wait for the access token instead? Can we await a cookie in puppeteer? Wouldn't that be more reliable i guess?

Reply:
I just saw you reply... I implemented it in another way now...If you fin that it doesn't work I'll wait for the cookies and then evaluate